resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -281,7 +281,7 @@ private[yarn] class YarnAllocator(
       val resourcesWithDefaults =
         ResourceProfile.getResourcesForClusterManager(rp.id, rp.executorResources,
           MEMORY_OVERHEAD_FACTOR, sparkConf, isPythonApp,
-          ResourceRequestHelper.resourceNameMapping)
+          ResourceRequestHelper.getResourceNameMapping(sparkConf))
```

Comment:
nit: Given this should not change, do we want to make it a field and reference that here ?